core/src/main/scala/com/avast/clients/rabbitmq/ConsumerBase.scala

```diff
@@ -88,18 +88,28 @@ private[rabbitmq] trait ConsumerBase[F[_]] extends StrictLogging {
   protected def createPropertiesForRepublish(newHeaders: Map[String, AnyRef],
                                              properties: BasicProperties,
                                              routingKey: String): BasicProperties = {
-    // values in newHeaders will overwrite values in original headers
-    // we must also ensure that UserID will be the same as current username (or nothing): https://www.rabbitmq.com/validated-user-id.html
     val originalUserId = Option(properties.getUserId).filter(_.nonEmpty)
-    val h = originalUserId match {
-      case Some(uid) => newHeaders + (RepublishOriginalRoutingKeyHeaderName -> routingKey) + (RepublishOriginalUserId -> uid)
-      case None => newHeaders + (RepublishOriginalRoutingKeyHeaderName -> routingKey)
+
+    val allNewHeaders = Map.newBuilder[String, AnyRef]
+
+    // copy original headers
+    allNewHeaders ++= Option(properties.getHeaders).map(_.asScala.map { case (k, v) => k.toLowerCase -> v }).getOrElse(Map.empty)
```

Comment:
This seems to be dangerous for me because it modifies all the headers names to lower-case so this could affect the consumers.